systemtest/src/test/java/io/strimzi/systemtest/operators/NamespaceDeletionRecoveryST.java

```diff
@@ -138,7 +142,17 @@ void testTopicNotAvailable() {
             LOGGER.info("Claim: {} has bounded Volume: {}", pvc.getMetadata().getName(), pv.getMetadata().getName());
         }
 
-        String kafkaPodName = kubeClient().listPodsByPrefixInName(testStorage.getNamespaceName(), testStorage.getBrokerComponentName()).get(0).getMetadata().getName();
+        // store cluster-id from Kafka CR
+        final String oldClusterId = KafkaResource.kafkaClient()
+            .inNamespace(testStorage.getNamespaceName())
+            .withName(testStorage.getClusterName())
+            .get()
+            .getStatus()
+            .getClusterId();
+
+        final Map<String, String> kafkaLabels = KafkaNodePoolResource.getLabelSelector(testStorage.getClusterName(), testStorage.getBrokerPoolName(), ProcessRoles.BROKER).getMatchLabels();
+        final String kafkaPodName = kubeClient().listPods(testStorage.getNamespaceName(), new LabelSelectorBuilder()
```

Comment:
Why you don't use the `KafkaNodePoolResource.getLabelSelector()` method directly? Why you are building the LabelSelector again?

Reply:
🤦 fixed. Thanks for pointing that out.